skfda/preprocessing/dim_reduction/feature_extraction/_ddg_transformer.py

```diff
@@ -0,0 +1,122 @@
+"""Feature extraction transformers for dimensionality reduction."""
+from typing import Sequence, Union
+
+import numpy as np
+from numpy import ndarray
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_is_fitted as sklearn_check_is_fitted
+
+from ...._utils import _classifier_fit_distributions
+from ....exploratory.depth import Depth, ModifiedBandDepth
+from ....representation.grid import FDataGrid
+
+default_depth = ModifiedBandDepth()
```

Comment:
What happens if two `DDGTransformer` objects with the default depth are used in two different threads at the same time (or in async contexts or whatever)?

Reply:
Whoops